File: src/main/java/com/team573/gongguri/domain/chat/entity/ChatRoomParticipation.java
package com.team573.gongguri.domain.chat.entity;

import com.team573.gongguri.domain.member.entity.Member;
import com.team573.gongguri.global.entity.BaseEntity;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@NoArgsConstructor
@Table(name = "chat_room_participation")
public class ChatRoomParticipation extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long chatRoomParticipantId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "member_id", nullable = false)
    private Member member;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "chat_room_id", nullable = false)
    private ChatRoom chatRoom;

    @Builder
    public ChatRoomParticipation(Member member, ChatRoom chatRoom) {
        this.member = member;
        this.chatRoom = chatRoom;
    }
}

File: src/main/java/com/team573/gongguri/domain/chat/entity/ChatRoomParticipation.kt
package com.team573.gongguri.domain.chat.entity

import com.team573.gongguri.domain.member.entity.Member
import com.team573.gongguri.global.entity.BaseEntity
import jakarta.persistence.*

@Entity
@Table(name = "chat_room_participation")
class ChatRoomParticipation (
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private val chatRoomParticipantId: Long? = null,

    @JoinColumn(name = "member_id", nullable = false)
    @ManyToOne(fetch = FetchType.LAZY)
    private var member: Member,

    @JoinColumn(name = "chat_room_id", nullable = false)
    @ManyToOne(fetch = FetchType.LAZY)
    private var chatRoom: ChatRoom
) : BaseEntity() {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as ChatRoomParticipation

        if (chatRoomParticipantId != other.chatRoomParticipantId) return false
        if (createdAt != other.createdAt) return false


        return true
    }

    override fun hashCode(): Int {
        var result = chatRoomParticipantId?.hashCode() ?: 0
        result = 31 * result + createdAt.hashCode()
        return result
    }
}


